packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -82,25 +98,100 @@ export class EditorSettingsSidebarComponent {
 	 * @returns {Promise<void>} No return value.
 	 */
 	async clickTab( tabName: EditorSidebarTab ): Promise< void > {
-		await this.frame.click( selectors.tabButton( tabName ) );
-		await this.frame.waitForSelector( selectors.activeTabButton( tabName ) );
+		const locator = this.frameLocator.locator( selectors.tabButton( tabName ) );
+		await locator.click();
+
+		const classAttributes = await locator.getAttribute( 'class' );
+		if ( ! classAttributes?.includes( 'is-active' ) ) {
+			throw new Error( `Failed to verify ${ tabName } is active.` );
+		}
 	}
 
 	/**
-	 * If the provided sidebar section is collapsed, toggles it to be expanded.
+	 * Expands a collapsed section of the sidebar.
 	 *
-	 * @param {EditorSidebarSection} sectionName Name of section.
+	 * If the section is already open, this method will pass.
+	 *
+	 * @param {ArticleSections} name Name of section to be expanded.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async expandSection( sectionName: EditorSidebarSection ): Promise< void > {
-		// Avoid the wpcalypso/staging banner
+	async expandSection( name: ArticleSections ): Promise< void > {
+		if ( await this.targetIsOpen( selectors.section( name ) ) ) {
+			return;
+		}
```

Comment:
I think this check works really well here, because it's not following any direct action, so I don't think we'll ever hit a race condition!